src/components/CSVButton/CSVButton.jsx

```diff
@@ -0,0 +1,258 @@
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    return [];
+  }
+
+  function mapResponse() {
+    if (type === 'allAreas') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'area') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'site') {
+      return studentResponseData.map(student => [
+        student.siteId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    return [];
+  }
+
+  const CSVReport = {
+    data: studentResponseData,
+    headers: createHeaders(),
+    filename: 'Areas_Report.csv',
+  };
+
+  const addAssociatedSiteToArea = async resData => {
+    async function fetchStudentsByAllAreas() {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const element of resData) {
+        TLPBackend.get(`/students/area/${element.areaId}`)
+          .then(res => {
+            res.data.forEach(student => {
+              // eslint-disable-next-line no-param-reassign
+              student.areaId = element.areaId;
+            });
+            if (res.data.length !== 0) {
+              studentResponseData.push(...res.data);
+            }
+          })
+          .catch(() => {
+            /* TODO document why this arrow function is empty */
+          });
+      }
+    }
+
+    async function fetchStudentsByArea() {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const element of resData) {
```

Comment:
Same question as above - if this function fetches students by area, shouldn't we just make one request to `/students/area/:areaId` based on the given `areaId`? Looping through all the given sites is going to be really time consuming